pkg/apis/openebs.io/scp/v1alpha1/storage-pool.go

```diff
@@ -17,20 +17,173 @@
 package v1alpha1
 
 import (
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	// StoragePoolVolumeProtectionFinalizer makes sure that no volume is present in the pool at the time of storage
+	// pool CR deletion.
+	StoragePoolVolumeProtectionFinalizer = "openebs.io/volume-protection"
+
+	// StoragePoolSourceProtectionFinalizer makes sure that internal pool resource is not present at the time of
+	// storage pool CR deletion.
+	StoragePoolSourceProtectionFinalizer = "openebs.io/pool-source-protection"
+)
+
 // +genclient
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 // +resource:path=storagepool
 
+// StoragePool describes a Storage Pool resource.
 // +kubebuilder:object:root=true
-// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:resource:scope=Namespaced,shortName=sp
 type StoragePool struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
+	// Spec is the specifications and configurations to be consumed by the provisioner for a StoragePool resource
+	// +required
+	Spec StoragePoolSpec `json:"spec"`
+	// Status is for handling status of StoragePool resource
+	// +optional
+	Status StoragePoolStatus `json:"status,omitempty"`
+}
+
+// StoragePoolSpec is the spec for a StoragePool resource
+type StoragePoolSpec struct {
+	// StorageCohort points to the StorageCohort resource, the pool is to be created upon
+	// +required
+	StorageCohort v1.ObjectReference `json:"storageCohort"`
+
+	// Type refers to the type of pool we are expecting to be created by the provisioner
+	// +required
+	Type string `json:"type"`
+
+	// Configuration points to the configuration, a custom resource, map of parameters or configmap
+	// that can be used to specify the pool and its device related configuration.
+	// +required
+	Configuration interface{} `json:"configuration"`
```

Comment:
We thought of use a reference here correct?

Reply:
Yes. But as per the last discussion it was decided that we should allow, map of strings as well so, to handle all the cases we decided to remove the deviceconfiguration and poolconfiguration and added them under this. cc @avishnu

Reply:
That's correct. A generic interface which can support all types of children like an object reference, a map of strings, lists, etc.